lib/hammer_cli_katello/capsule.rb

```diff
@@ -133,6 +133,105 @@ class SyncCommand < HammerCLIForemanTasks::AsyncCommand
         build_options
       end
 
+      class SyncStatusCommand < HammerCLIKatello::InfoCommand
+        resource :capsule_content, :sync_status
+        command_name "synchronization-status"
+
+        output do
+          field :last_sync_time, _('Last sync'), Fields::Date
+          field :_status, _('Status')
+          collection :active_sync_tasks, _('Currently running sync tasks') do
+            field :id, _('Task id')
+            field :_progress, _('Progress')
+          end
+          label _('Last failure') do
+            from :_last_failure do
+              field :id, _('Task id')
+              from :humanized do
+                field :errors, _('Messages'), HammerCLIKatello::Output::Fields::Array
+              end
+            end
+          end
+        end
+
+        private
+
+        def extend_data(data)
+          data["_status"] = sync_status(data)
+          data['_last_failure'] = data["last_failed_sync_tasks"][-1]
+
+          data["active_sync_tasks"].each do |task|
+            task['_progress'] =  format_progress(task['progress'])
+          end
+          data
+        end
+
+        def sync_status(data)
+          syncable_envs = data["lifecycle_environments"].select { |env| env['syncable'] }
+          if syncable_envs.empty?
+            _("Capsule is synchronized")
+          else
+            env_names = syncable_envs.map { |env| env['name'] }.join(", ")
+            _("%{count} environment(s) can be synchronized: %{names}") % {
+              :count => syncable_envs.size,
+              :names => env_names
+            }
+          end
+        end
+
+        def format_progress(progress)
+          "#{(progress * 100).floor}%"
+        end
+
+        build_options
+      end
+
+      class InfoCommand < HammerCLIKatello::InfoCommand
+        resource :capsule_content, :sync_status
+        command_name "info"
+
+        output do
+          collection :lifecycle_environments, _("Environments") do
+            field :name, _('Name')
+            field :organization, _("Organization"), Fields::Reference
+            collection :content_views, _("Content Views") do
+              field nil, _("Name"), Fields::Reference
+              field :composite, _('Composite'), Fields::Boolean
+              field :last_published, _('Last Published'), Fields::Date
+              label _('Content') do
+                from :counts do
+                  field :content_hosts, _('Hosts')
+                  field :products, _('Products')
+                  field :yum_repositories, _('Yum repos')
+                  field :docker_repositories, _('Docker repos')
+                  field :packages, _('Packages')
+                  field :package_groups, _('Package groups')
+                  field :errata, _('Errata')
+                  field :puppet_modules, _('Puppet modules')
+                end
+              end
+            end
+          end
+        end
+
+        def extend_data(data)
```

Comment:
Is overriding this method necessary since you aren't manipulating the data in any way?